app/views/repositories/show.html.erb

```diff
@@ -9,19 +9,28 @@
 <section id="show-repository">
 
   <section id="repository-photos">
-
-
+    <% if @photos.length != 0 %>
       <div id="show-photo">
         <%= image_tag @photos.first.try(:image).try(:url), data: { width: @photos.first.width, height: @photos.first.height} %>
       </div>
-
-
     <div id="photo-slide">
       <% @photos.each do |p| %>
-        <%= image_tag p.image.url, data: { width: p.width, height: p.height} %>
+        <!-- is this 'if p.valid?' necessary?(I added it to make sure an invalid photo doesn't cause error) -->
+        <% if p.valid? %>
```

Comment:
I don't think so. The problem is usually repositories without images, not repositories with invalid images